src/plugin/memory-webview-main.ts

```diff
@@ -208,6 +213,10 @@ export class MemoryWebview implements vscode.CustomReadonlyEditorProvider {
         return participant;
     }
 
+    protected setTitle(webviewParticipant: WebviewIdMessageParticipant, title: string): void {
+        this.messenger.sendNotification(setTitleType, webviewParticipant, title);
+    }
```

Comment:
The machinery to set title from plugin -> webview feels a little heavy: at the moment, it's only used on panel initialization (and may not be working correctly, since in my test, the panel showed the default `Memory` rather than `Memory + Index`).

On the other hand I don't actually have a suggestion here - see my next comment.

Reply:
Right, it indeed is a bit heavy-weight. But I'm not aware of a mechanism to store state across webview instances other than in the extension "main".